### ISSUE TITLE
关于论文与code的问题

### ISSUE THREAD

**Issue by nishiwen1214** [chunyuanY/RumorDetection#11]
1、论文中有提到“build the connection inside between source microblog and retweet”但是code中并没有这一步
code：
X_text = self.mh_attention(X_text, X_text, X_text)
传进去的参数都是原始贴的文本
2、论文中有提到使用用户特征，但是数据里好像没有用户特征，请问是怎么利用用户特征的？

**Comment by DayDreamChaser** [comment]
**代码中和论文不一致的地方**：

1.  只使用了source tweet的文本，对source tweet的词序列做了多头注意力操作
（并没有retweet和source tweet的多头注意力等一系列融合操作）
  ` X_text = self.word_embedding(X_text) `  
  ` X_text = self.mh_attention(X_text, X_text, X_text) `

    source tweet的嵌入是通过**对做了多头注意力source tweet的词嵌入做卷积操作得到的**。

2.  在编码全局关系阶段，代码中只有对如下待学习的user_tweet_embedding的结点嵌入的GAT操作，
     `self.user_tweet_embedding = nn.Embedding(self.uV, 300, padding_idx=0)`

     并没有**user embedding**和**source tweet embedding**的之间的交替的注意力操作。 
    ` X = self.user_tweet_embedding(torch.arange(0, self.uV).long().to(device)) `
    ` X = self.dropout(X)   `
    ` X = torch.cat([att(X, self.adj) for att in self.attentions], dim=1) `
    ` X = self.dropout(X) `
    ` X = F.elu(self.out_att(X, self.adj)) `
    ` X_ = X[X_tid]  `
